### PR TITLE
Camera feature

### PR DESCRIPTION
**Pull request by jbennyhuang**
159171584_432477011185107_6307530491577010016_n

Camera stuff

### CODE REVIEW

File: navigation/BottomTabNavigator.tsx [hunk 1]
```diff
@@ -71,3 +80,17 @@ function TabTwoNavigator() {
...
+  return (
+    <TabCameraStack.Navigator>
+      <TabCameraStack.Screen
+        name="CameraScreen"
+        component={CameraScreen}
+        options={{ headerTitle: 'Camera', headerShown: false }}
+      />
+    </TabCameraStack.Navigator>
+  );
+}
```
Comment (jx3yang) [inline navigation/BottomTabNavigator.tsx]: newline

File: screens/CameraScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
+import * as React from 'react';
+import { useFocusEffect } from '@react-navigation/native';
+import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
+import { Camera } from 'expo-camera';
+import { Ionicons, MaterialIcons, MaterialCommunityIcons } from '@expo/vector-icons';
+
+export default function CameraScreen({ navigation: { goBack } }) {
+
+  const [camera, setCamera] = React.useState<Camera | null>(null);
```
Comment (jx3yang) [inline screens/CameraScreen.tsx]: `useState<Camera>()` will set default `camera` to `undefined`, no need to explicitly use `null`.

File: screens/CameraScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,181 @@
+import * as React from 'react';
+import { useFocusEffect } from '@react-navigation/native';
+import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
+import { Camera } from 'expo-camera';
+import { Ionicons, MaterialIcons, MaterialCommunityIcons } from '@expo/vector-icons';
+
+export default function CameraScreen({ navigation: { goBack } }) {
```
Comment (jx3yang) [inline screens/CameraScreen.tsx]: probably want a type on the props

File: screens/CameraScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+  const [camera, setCamera] = React.useState<Camera | null>(null);
+  const [hasPermission, setHasPermission] = React.useState<any>(null);
```
Comment (jx3yang) [inline screens/CameraScreen.tsx]: `useState<boolean>()`, and check for `hasPermission === undefined` on line 28

File: screens/CameraScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+  const [type, setType] = React.useState<any>(Camera.Constants.Type.back);
```
Comment (jx3yang) [inline screens/CameraScreen.tsx]: probably `useState<Camera.Constants.Type>(Camera.Constants.Type.back)` if `Camera.Constants.Type` is an enum

File: screens/CameraScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+                setType(
+                  type === Camera.Constants.Type.back
+                    ? Camera.Constants.Type.front
+                    : Camera.Constants.Type.back
+                );
+              }}>
+              { type === Camera.Constants.Type.back ?  
+                <Ionicons name='camera-reverse' size={35} color='white' /> :
+                <Ionicons name='camera-reverse-outline' size={35} color='white' />
+              }
```
Comment (jx3yang) [inline screens/CameraScreen.tsx]: if the name is the only thing that is changing, perhaps
```
<Ionicons
  name={ type === Camera.Constants.Type.back ? 'camera-reverse' : 'camera-reverse-outline' }
  ...
/>
```

File: types.tsx [hunk 7]
```diff
@@ -17,3 +18,7 @@ export type TabOneParamList = {
 export type TabTwoParamList = {
   TabTwoScreen: undefined;
 };
+
+export type TabCameraParamList = {
+  CameraScreen: undefined;
+}
```
Comment (jx3yang) [inline types.tsx]: newline

**Review by jx3yang — CHANGES_REQUESTED** [review]

File: screens/CameraScreen.tsx (hunk 5 above)
Comment (jbennyhuang) [reply screens/CameraScreen.tsx]: Type is any in the source code

image

File: screens/CameraScreen.tsx (hunk 2 above)
Comment (jbennyhuang) [reply screens/CameraScreen.tsx]: It gives me this error if I only use <Camera> so it must be <Camera | null> but you're right I don't need to explicitly use null as the initial value.

image

File: screens/CameraScreen.tsx (hunk 5 above)
Comment (jx3yang) [reply screens/CameraScreen.tsx]: if the only types that we care about are `back` and `front`, you could make a custom type (or enum)
```
type CameraType = Camera.Constants.Type.back | Camera.Constants.Type.front;
```
and use `useState<CameraType>(Camera.Constants.Type.back)`

this provides a bit more transparency

File: screens/CameraScreen.tsx (hunk 2 above)
Comment (jx3yang) [reply screens/CameraScreen.tsx]: I see, then you can leave it as it was, or write `useState<Camera>()` and `setCamera(ref || undefined)`

File: screens/CameraScreen/index.tsx [hunk 8]
```diff
@@ -179,3 +179,5 @@ const styles = StyleSheet.create({
     color: 'white',
   },
 });
+
+export default CameraScreen;
```
Comment (jx3yang) [inline screens/CameraScreen/index.tsx]: newline

File: screens/CameraScreen.tsx (hunk 5 above)
Comment (jbennyhuang) [reply screens/CameraScreen.tsx]: alright sounds good

**Review by jx3yang — APPROVED** [review]

**Pull request merged**